_inc/lib/core-api/wpcom-endpoints/subscribers.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Subscribers: Get subscriber count
+ *
+ * @since 6.9
+ */
+class WPCOM_REST_API_V2_Endpoint_Subscribers extends WP_REST_Controller {
+	function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'subscribers';
+		// This endpoint *does not* need to connect directly to Jetpack sites.
+		$this->wpcom_is_wpcom_only_endpoint = true;
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	public function register_routes() {
+		// GET /sites/<blog_id>/subscribers/count - Return number of subscribers for this site.
+		register_rest_route( $this->namespace, '/' . $this->rest_base  . '/count', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_subscriber_count' ),
+				'permission_callback' => array( $this, 'readable_permission_check' ),
+			)
+		) );
+	}
+
+	public function readable_permission_check() {
+		if ( ! current_user_can_for_blog( get_current_blog_id(), 'edit_posts' ) ) {
+			return new WP_Error( 'authorization_required', 'Only users with the permission to edit posts can see the subscriber count.', array( 'status' => 401 ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves subscriber count
+	 *
+	 * @param WP_REST_Request $request incoming API request info
+	 * @return array data object containing subscriber count
+	 */
+	public function get_subscriber_count( $request ) {
+		$subscriptions = new Jetpack_Subscriptions_Widget();
```

Comment:
Since you are calling a class that depends on the Subscriptions module, you may run into the same problem that happened in #10727? I wonder if it may be worth adding a check, like in #10745?

Reply:
Thanks, I'd meant to update this to use the code to fetch the subscriber count from wpcom directly. I'll do that now.